README.adoc

```diff
@@ -0,0 +1,110 @@
+status]]
+
+:toc:
+:toc-placement!:
+
+Landrush is a simple cross-platform DNS for Vagrant VMs that is visible
+on both, the guest and the host.
+
+It spins up a small DNS server and redirects DNS traffic from your VMs
+to use it, automatically registering/unregistering IP addresses of
+guests as they come up and go down.
+
+NOTE: Windows support is currently considered experimental. If you
+face problems using Landrush on Windows please let us know.
+
+
+'''
+toc::[]
+'''
+
+== Installation
+
+Install under Vagrant (1.1 or later):
+
+....
+$ vagrant plugin install landrush
+....
+
+== Getting started
+
+.  Enable the plugin in your `Vagrantfile`:
++
+....
+config.landrush.enabled = true
+....
+.  Bring up a machine.
++
+....
+$ vagrant up
+....
+.  You are able to get your VM's hostname resolved on your host:
++
+....
+$ dig -p 10053 @localhost myhost.vagrant.test
+....
+.  If you shut down your guest, the entries associated with it will be
+removed.
+
+Landrush retrieves your VM's hostname from either the vagrant config or
+it uses the system's actual hostname by running the `hostname` command.
+A default hostname of "guest-vm" is assumed if the hostname is otherwise not
+available.
+
+A Landrush example configuration could look like this:
+
+....
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/precise64"
+
+  config.landrush.enabled = true
+
+  config.vm.hostname = "myhost.vagrant.test"
+
+  config.landrush.host 'static1.example.com', '1.2.3.4'
+  config.landrush.host 'static2.example.com', '2.3.4.5'
+end
+....
+
+See the link:Usage.adoc[Usage document] for further information on Landrush usage.
+
+== Available CLI commands
+
+Check out `vagrant landrush help` for the available commands.
+
+....
+vagrant landrush <command>
+
+commands:
+  {start|stop|restart|status}
+    control the landrush server daemon
+  list, ls
+    list all DNS entries known to landrush
+  dependentvms, vms
+    list vms currently dependent on the landrush server
+  set { <host> <ip> | <alias> <host> }
+    adds the given host-to-ip or alias-to-hostname mapping.
+    Existing host ip addresses will be overwritten
+  rm, del { <host> | <alias> | --all }
+    delete the given hostname or alias from the server.
+     --all removes all entries
+  help
+    you're lookin at it!
+....
+
+See the link:Troubleshooting.adoc[Troubleshooting document] to resolve issues you face while using Landrush.
+
+== Help Out!
+
+This project could use your feedback and help! Please don't hesitate to
+open issues or submit pull requests. NO HESITATION IS ALLOWED. NONE
+WHATSOEVER. +
+If you wish to contribute to the development of Landrush, refer to the link:Development.adoc[Development document] for tips on getting started. See the link:CONTRIBUTING.md[Contributing document] for details on how you can contribute.
```

Comment:
It seems odd that the sentence starts with "if you want to contribute", but the link is to the develop document. I think we should switch this.

"If you wish to contribute to the development of Landrush, refer to the Contributing guide to get started. The Development guide will then help you to setup your development environment."

BTW, I prefer the term 'guide' over 'document'. The latter sounds so formal.